### DISCUSSION TITLE
Whats the correct flow for doing a Family registration ?

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#1452] [Q&A]
What are your thoughts on this UX?

Current Implementation 

1. Click on Household register 
2. Takes you to the  Household list 
3. Add a family 
4. Goes back to household list 

As per @HenryRae review 

1. Add family
2. On success go to a family profile page
3. On add family member, add a new family member in the family member list in the family profile and show a snack bar too

`We don't need to go to a family member's profile page after creation since CHA might be in the process of adding several family members. The reason why we navigate to family profiles after creation is to encourage them to add family members and also inform them, that the family has been created but has no family members`

**Comment by rowo** [comment]
This also depends on if the HH head is required and if you can add members in the family registration flow — those decisions and because the "add family" button is in the menu, means going to the family profile makes sense for eCBIS.

In general though, this change happened because CHWs using VTS 1.0 were getting lost when after adding a family. The most common behavior after adding a family was adding a member and with the old approach we forced people to find the family and navigate to it. 

So I don't know if this will always be the pattern across all FHIR apps, but it makes sense here with the family-based CHW apps.

**Reply by pld** [reply]
Cool, makes sense for this to be configurable, where you go to after creating a family. The general product feature here is being able to configure where you go after submitting a Questionnaire

**Reply by dubdabasoduba** [reply]
@pld @f-odhiambo @rowo This is the idea in the navigation config and will be the idea on the profile menu configs. Every place we load a form will have an `onClose` action which describes what happens after a form is saved. 

We will have default behaviour but the user can override that by changing the `onClose` reference.

**Reply by pld** [reply]
Yea so I think the answer to this discussion is that there is no "correct" flow for a family registration, and specifically doing a family registration is not a workflow that should be built in to the product.

The product should enable content creators to use questionnaires and configs to implement whatever workflow is correct for their process, and we will have to iterate on what configs support to handle variations in how content creators would like their workflows to function.
